pkg/registry/apps/statefulset/strategy.go

```diff
@@ -73,6 +88,20 @@ func (statefulSetStrategy) PrepareForCreate(ctx context.Context, obj runtime.Obj
 	statefulSet.Generation = 1
 
 	pod.DropDisabledTemplateFields(&statefulSet.Spec.Template, nil)
+	dropStatefulSetDisabledFields(statefulSet, nil)
+}
+
+// maxUnavailableInUse returns true if StatefulSet's maxUnavailable set(used)
+func maxUnavailableInUse(statefulset *apps.StatefulSet) bool {
+	if statefulset == nil {
+		return false
+	}
+	if statefulset.Spec.UpdateStrategy.RollingUpdate != nil {
```

Comment:
for consistency, do the same you are doing above:

```golang
if statefulset.Spec.UpdateStrategy.RollingUpdate == nil {
  return false
}
return statefulset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable != nil
}

Reply:
done